wordpress-varnish-as-a-service.php

```diff
@@ -498,139 +588,51 @@ function WPVarnishPurgeObject($wpv_url_obj) {
-					}
-				}
-			} else {
-				if ($wpv_use_version == 3) {
-					$varnish_test_conn .= "<li><span style=\"color: blue;\">".__("INFO - HTTP BAN",$this->plugin_group)."</span></li>\n";
-					$out = "BAN HTTP/1.0\r\n";
-					$out .= "X-Ban-Url: $wpv_url\r\n";
-					$out .= "X-Ban-Host: $wpv_host\r\n";
-					$out .= "Connection: Close\r\n\r\n";
-					fwrite($varnish_sock, $out."\n");
-					$buf = fread($varnish_sock, 256);
-					if(preg_match('/200/', $buf)) {
-						$varnish_test_conn .= "<li><span style=\"color: green;\">".__("OK - Cache flush",$this->plugin_group)."</span></li>\n";
-					} else {
-						$varnish_test_conn .= "<li><span style=\"color: red;\">".__("KO - Cache flush",$this->plugin_group)."</span><br><small>".__("Verify your Varnish version",$this->plugin_group)."</small></li>\n";
-					}
 
-				} else {
-					$varnish_test_conn .= "<li><span style=\"color: blue;\">".__("INFO - HTTP Purge",$this->plugin_group)."</span></li>\n";
-					$out = "PURGE $wpv_url HTTP/1.0\r\n";
-					$out .= "Host: $wpv_host\r\n";
-					$out .= "Connection: Close\r\n\r\n";
-					fwrite($varnish_sock, $out);
-					$buf = fread($varnish_sock, 256);
-					if(preg_match('/200 OK/', $buf)) {
-						$varnish_test_conn .= "<li><span style=\"color: green;\">".__("OK - Request",$this->plugin_group)."</span></li>\n";
 					} else {
-						$varnish_test_conn .= "<li><span style=\"color: red;\">".__("KO - Request",$this->plugin_group)."</span></li>\n";
+						$varnish_test_conn .= "<li><span style=\"color: blue;\">".__("INFO - HTTP Purge",$this->plugin_group)."</span></li>\n";
+						$out = "PURGE $wpv_url HTTP/1.0\r\n";
+						$out .= "Host: $wpv_host\r\n";
+						$out .= "Connection: Close\r\n\r\n";
+						fwrite($varnish_sock, $out);
+						$buf = fread($varnish_sock, 256);
+						if(preg_match('/200 OK/', $buf)) {
+							$varnish_test_conn .= "<li><span style=\"color: green;\">".__("OK - Request",$this->plugin_group)."</span></li>\n";
+						} else {
+							$varnish_test_conn .= "<li><span style=\"color: red;\">".__("KO - Request",$this->plugin_group)."</span></li>\n";
+						}
 					}
 				}
+				fclose($varnish_sock);
+			} else {
+				$varnish_test_conn .= "<li><span style=\"color: red;\">".__("KO - Connection to Server",$this->plugin_group)."</span><br><small>".__("IP address or port closed. Verify your firewall or iptables.",$this->plugin_group)."</small></li>\n";
 			}
-			fclose($varnish_sock);
-		} else {
-			$varnish_test_conn .= "<li><span style=\"color: red;\">".__("KO - Connection to Server",$this->plugin_group)."</span><br><small>".__("IP address or port closed. Verify your firewall or iptables.",$this->plugin_group)."</small></li>\n";
+			$varnish_test_conn .= "</ul>\n";
+			?>
+			<div class="updated"><?php echo $varnish_test_conn; ?></div>
+			<?php
 		}
-		$varnish_test_conn .= "</ul>\n";
-		?>
-		<div class="updated"><?php echo $varnish_test_conn; ?></div>
-		<?php
 	}
 }
```

Comment:
if the `WPVarnish` class already exists (e.g. a different version of the plugin is installed and activated), should we be instantiating a new `WPVarnish` object at the bottom of this file at all?

Reply:
Moved admin if statement into the class exists if statement.